php/tests.php

```diff
@@ -41,4 +41,58 @@ public function testAddParagraphsMalformed(){
         addParagraphs($input);
     }
 
+
+    public function testEditParagraphDropdownSuccess(){
+        $expected = '<option value=>paragraph 1 Hello World</option>';
+        $input = [['paragraph'=>'Hello World']];
+        $case = editParagraphDropdown($input);
+        $this->assertEquals($expected, $case);
+    }
+
+    public function testEditParagraphDropdownFailure(){
+        $expected = '';
+        $input = [['paragraph' => []]];
+        $case = editParagraphDropdown($input);
+        $this->assertEquals($expected, $case);
+    }
+
+    public function testEditParagraphDropdownFailure2(){
+        $expected = '';
+        $input = [['awesome' => []]];
+        $case = editParagraphDropdown($input);
+        $this->assertEquals($expected, $case);
+    }
+
+    public function testEditParagraphDropdownMalformed(){
+        $input = 3;
+        $this->expectException(TypeError::class);
+        editParagraphDropdown($input);
+    }
+
+    public function testPasteEditSuccess(){
```

Comment:
It looks like your function does the same thing whether the case is a success or failure which doesn't seem right?